modules/custom/activity_basics/activity_basics.module

```diff
@@ -14,13 +15,32 @@ use Drupal\group\Entity\GroupContent;
  */
 function activity_basics_entity_insert(EntityInterface $entity) {
   _activity_basics_entity_action($entity, 'create_entitiy_action');
+
+  // Check if the entity created with "published" status.
+  if ($entity instanceof EntityPublishedInterface) {
+    if ($entity->isPublished()) {
+      _activity_basics_entity_action($entity, 'publish_entity_action');
+    }
+  }
 }
 
 /**
  * Implements hook_entity_update().
  */
 function activity_basics_entity_update(EntityInterface $entity) {
   _activity_basics_entity_action($entity, 'update_entity_action');
+
+  // Check if the entity become published.
+  if ($entity instanceof EntityPublishedInterface) {
+    // Check if entity change state from "unpublished" to "published".
+    // @phpstan-ignore-next-line
+    $original = $entity->original ?? NULL;
```

Comment:
Move changes in this hook to the "entity_presave" hook where the "original" property will be always available.

Reply:
Fixed